### PR TITLE
Finish Static Pages

### PR DESCRIPTION
**Pull request by hitshootnum1**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,47 +1,34 @@
...
 gem 'turbolinks'
-# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.0'
-# bundle exec rake doc:rails generates the API under doc/api.
-gem 'sdoc', '~> 0.4.0', group: :doc
 
-# Use ActiveModel has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
+# Bootstrap
+gem 'will_paginate',           '3.0.7'
```
Comment (hellovietnam93) [inline Gemfile]: using "" instead of '', edit for all below

File: Gemfile [hunk 2]
```diff
@@ -1,47 +1,34 @@
...
-# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
 gem 'turbolinks'
-# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.0'
-# bundle exec rake doc:rails generates the API under doc/api.
-gem 'sdoc', '~> 0.4.0', group: :doc
 
-# Use ActiveModel has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
+# Bootstrap
```
Comment (hellovietnam93) [inline Gemfile]: remove all remark

File: Gemfile [hunk 3]
```diff
@@ -1,47 +1,34 @@
...
+gem 'will_paginate',           '3.0.7'
+gem 'bootstrap-will_paginate', '0.0.10'
+gem 'bootstrap-sass',          '3.2.0.0'
```
Comment (hellovietnam93) [inline Gemfile]: using only one space after ```,```

File: app/assets/javascripts/static_pages.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (hellovietnam93) [inline app/assets/javascripts/static_pages.coffee]: delete this file

File: app/assets/stylesheets/custom.css.scss [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+/* miscellaneous */
+
+.debug_dump {
+  clear: both;
+  float: left;
+  width: 100%;
+  margin-top: 45px;
+  @include box_sizing;
+}
```
Comment (hellovietnam93) [inline app/assets/stylesheets/custom.css.scss]: lack of endfile

File: app/assets/stylesheets/static_pages.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (hellovietnam93) [inline app/assets/stylesheets/static_pages.scss]: delete this file

File: app/helpers/application_helper.rb [hunk 7]
```diff
@@ -1,2 +1,12 @@
 module ApplicationHelper
+
+	# Return the full title on a per-page basis.
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: remove remark

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -1,2 +1,12 @@
...
+	def full_title page_title = ''
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: ident

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -1,2 +1,12 @@
...
+		base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: "Ruby on Rails Tutorial Sample App" to I18n

File: app/helpers/static_pages_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (hellovietnam93) [inline app/helpers/static_pages_helper.rb]: delete this file

File: app/views/layouts/_footer.html.erb [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    The <a href="#">E-Learning Program</a>
```
Comment (hellovietnam93) [inline app/views/layouts/_footer.html.erb]: I18n

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -1,14 +1,22 @@
...
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= full_title(yield(:title)) %></title>
+    <%= stylesheet_link_tag    'application', media: 'all', 
+                                              'data-turbolinks-track' => true %>
```
Comment (hellovietnam93) [inline app/views/layouts/application.html.erb]: ident

File: config/routes.rb [hunk 13]
```diff
@@ -1,56 +1,6 @@
...
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  
+  root                'static_pages#home'
```
Comment (hellovietnam93) [inline config/routes.rb]: one space

File: Gemfile [hunk 14]
```diff
@@ -1,47 +1,28 @@
...
+gem "turbolinks"
+gem "jbuilder", "~> 2.0"
+gem "will_paginate", "3.0.7"
+gem "bootstrap-will_paginate", "0.0.10"
+gem "bootstrap-sass", "3.2.0.0"
+gem "bcrypt", "3.1.7"
+gem "faker", "1.4.2"
+gem "carrierwave", "0.10.0"
+gem "mini_magick", "3.8.0"
+
```
Comment (hellovietnam93) [inline Gemfile]: remove this line

File: app/assets/stylesheets/custom.css.scss [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+@import "bootstrap";
+
+/* mixins, variables, etc. */
+
+$gray-medium-light: #eaeaea;
+
+@mixin box_sizing {
+  -moz-box-sizing:    border-box;
+  -webkit-box-sizing: border-box;
+  box-sizing:         border-box;
```
Comment (hellovietnam93) [inline app/assets/stylesheets/custom.css.scss]: one space agian, plz check for all

File: app/helpers/application_helper.rb [hunk 16]
```diff
@@ -1,2 +1,11 @@
 module ApplicationHelper
+
+  def full_title page_title = ''
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: using ""

File: app/helpers/application_helper.rb [hunk 17]
```diff
@@ -1,2 +1,11 @@
...
+    base_title = I18n.t 'base_title'
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: in helper only need  base_title = t "base_title"

File: app/views/layouts/_footer.html.erb [hunk 18]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    The <a href="#"><%= t('application_name') %></a>
```
Comment (hellovietnam93) [inline app/views/layouts/_footer.html.erb]: I18n again

File: app/views/layouts/_footer.html.erb [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+    by <a href="#">Framgia</a>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to "About",   "#" %></li>
+      <li><%= link_to "Contact", "#" %></li>
+      <li><a href="#">News</a></li>
```
Comment (hellovietnam93) [inline app/views/layouts/_footer.html.erb]: I18n

File: app/views/layouts/_footer.html.erb [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+    </ul>
+  </nav>
+</footer>
```
Comment (hellovietnam93) [inline app/views/layouts/_footer.html.erb]: lack of end file

File: app/views/layouts/application.html.erb [hunk 21]
```diff
@@ -1,14 +1,22 @@
...
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
-  <%= csrf_meta_tags %>
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= full_title(yield(:title)) %></title>
```
Comment (hellovietnam93) [inline app/views/layouts/application.html.erb]: remove ()

File: app/views/layouts/_footer.html.erb [hunk 22]
```diff
@@ -0,0 +1,12 @@
+<footer class="footer">
+  <small>
+    <%= t("The") %> <a href="#"><%= t("application_name") %></a>
```
Comment (hellovietnam93) [inline app/views/layouts/_footer.html.erb]: lower case in I18n

File: app/views/layouts/_footer.html.erb [hunk 23]
```diff
@@ -0,0 +1,12 @@
...
+    <%= t("by") %> <a href="#">Framgia</a>
```
Comment (hellovietnam93) [inline app/views/layouts/_footer.html.erb]: I18n for Framgia,

dont need () like <%= t "by" %>

File: app/views/layouts/_header.html.erb [hunk 24]
```diff
@@ -0,0 +1,13 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "E-Learning", root_path, id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home",   root_path %></li>
```
Comment (hellovietnam93) [inline app/views/layouts/_header.html.erb]: one space

File: app/views/layouts/_header.html.erb [hunk 25]
```diff
@@ -0,0 +1,13 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "E-Learning", root_path, id: "logo" %>
```
Comment (hellovietnam93) [inline app/views/layouts/_header.html.erb]: I18n

File: app/views/layouts/_footer.html.erb [hunk 26]
```diff
@@ -0,0 +1,12 @@
+<footer class="footer">
+  <small>
+    <%= t "The" %> <a href="#"><%= t "application_name" %></a>
+    <%= t "by" %> <a href="#">Framgia</a>
```
Comment (vuducluan) [inline app/views/layouts/_footer.html.erb]: I18n (Framgia)

File: config/locales/en.yml [hunk 27]
```diff
@@ -1,23 +1,16 @@
...
+  The: "The"
+  by: "by"
+  Home: "Home"
+  Help: "Help"
+  Users: "Help"
+  Log_in: "Log in"
+  get_help: "Get help E-Learning application"
+  welcome: "Welcome to the E-Learning app"
+  home_page_welcome: "This is the home page for the E-Learning Application"
+  sign_up_now: "Sign up now!"
```
Comment (vuducluan) [inline config/locales/en.yml]: insert a blank line

File: config/routes.rb [hunk 28]
```diff
@@ -1,56 +1,6 @@
...
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  
+  root 'static_pages#home'
```
Comment (vuducluan) [inline config/routes.rb]: use "", not  ''

File: app/views/static_pages/home.html.erb [hunk 29]
```diff
@@ -0,0 +1,9 @@
+<div class="center jumbotron">
+	<h1><%= t "welcome" %></h1>
+
+	<h2>
+		<%= t "home_page_welcome" %>
```
Comment (vuducluan) [inline app/views/static_pages/home.html.erb]: indent

File: app/views/layouts/_footer.html.erb [hunk 30]
```diff
@@ -0,0 +1,12 @@
+<footer class="footer">
+  <small>
+    <%= t "The" %> <a href="#"><%= t "application_name" %></a>
```
Comment (hellovietnam93) [inline app/views/layouts/_footer.html.erb]: chỉ viết thường thôi nhé

File: app/views/layouts/_footer.html.erb [hunk 31]
```diff
@@ -0,0 +1,12 @@
...
+    <%= t "by" %> <a href="#"><%= t "Framgia" %></a>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to t("About"),   "#" %></li>
```
Comment (hellovietnam93) [inline app/views/layouts/_footer.html.erb]: 1 dấu cách

File: app/views/layouts/_header.html.erb [hunk 32]
```diff
@@ -0,0 +1,13 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to t("E_Learning"), root_path, id: "logo" %>
```
Comment (hellovietnam93) [inline app/views/layouts/_header.html.erb]: viết thường

File: app/views/layouts/_header.html.erb [hunk 33]
```diff
@@ -0,0 +1,13 @@
...
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to t("Home"),   root_path %></li>
```
Comment (hellovietnam93) [inline app/views/layouts/_header.html.erb]: 1 dấu cách

File: app/views/layouts/_header.html.erb [hunk 34]
```diff
@@ -0,0 +1,13 @@
...
+        <li><%= link_to t("Help"),   help_path %></li>
```
Comment (hellovietnam93) [inline app/views/layouts/_header.html.erb]: 1 dấu cách

File: app/views/layouts/_header.html.erb [hunk 35]
```diff
@@ -0,0 +1,13 @@
...
+        <li><%= link_to t("Users"), "#" %></li>
```
Comment (hellovietnam93) [inline app/views/layouts/_header.html.erb]: viết thường

File: app/views/layouts/_header.html.erb [hunk 36]
```diff
@@ -0,0 +1,13 @@
...
+        <li><%= link_to t("Log_in"), "#" %></li>
```
Comment (hellovietnam93) [inline app/views/layouts/_header.html.erb]: viết thường

File: config/application.rb [hunk 37]
```diff
@@ -22,5 +22,6 @@ class Application < Rails::Application
 
     # Do not swallow errors in after_commit/after_rollback callbacks.
     config.active_record.raise_in_transactional_callbacks = true
+    config.i18n.load_path += Dir[Rails.root.join('config', 'locales', '**', '*.{rb,yml}').to_s]
```
Comment (hellovietnam93) [inline config/application.rb]: sử dụng ""

File: app/views/layouts/application.html.erb [hunk 38]
```diff
@@ -1,14 +1,22 @@
...
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= full_title yield :title %></title>
+    <%= stylesheet_link_tag "application", media: "all",
+      "data-turbolinks-track" => true %>
+    <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
```
Comment (tqtrung09) [inline app/views/layouts/application.html.erb]: Convention.
"data-turbolinks-track": true

**Pull request merged**